diesel/src/query_builder/insert_statement.rs

```diff
@@ -24,26 +27,64 @@ impl<T, Op> IncompleteInsertStatement<T, Op> {
     }
 
     /// Specify which table the data passed to `insert` should be added to.
-    pub fn into<S>(self, target: S) -> InsertStatement<S, T, Op> where
-        InsertStatement<S, T, Op>: AsQuery,
+    pub fn into<S>(self, target: S) -> T::InsertStatement where
+        T: IntoInsertStatement<S, Op, NoReturningClause>,
     {
-        InsertStatement {
-            operator: self.operator,
+        self.records.into_insert_statement(target, self.operator, NoReturningClause)
+    }
+}
+
+pub trait IntoInsertStatement<Tab, Op=Insert, Ret=NoReturningClause> {
+    type InsertStatement;
+
+    fn into_insert_statement(self, target: Tab, operator: Op, returning: Ret)
+        -> Self::InsertStatement;
+}
+
+impl<'a, T, Tab, Op, Ret> IntoInsertStatement<Tab, Op, Ret> for &'a [T] {
+    type InsertStatement = BatchInsertStatement<Tab, Self, Op, Ret>;
+
+    fn into_insert_statement(self, target: Tab, operator: Op, returning: Ret)
+        -> Self::InsertStatement
+    {
+        BatchInsertStatement {
+            operator: operator,
             target: target,
-            records: self.records,
-            returning: NoReturningClause,
+            records: self,
+            returning: returning,
         }
     }
 }
 
+impl<'a, T, Tab, Op, Ret> IntoInsertStatement<Tab, Op, Ret> for &'a Vec<T> {
+    type InsertStatement = <&'a [T] as IntoInsertStatement<Tab, Op, Ret>>::InsertStatement;
+
+    fn into_insert_statement(self, target: Tab, operator: Op, returning: Ret)
+        -> Self::InsertStatement
+    {
+        (&**self).into_insert_statement(target, operator, returning)
+    }
+}
```

Comment:
Do you actually need to impl this for `&Vec` if you already have `&[]`? I mean, it doesn't hurt, but deref coercions should take care of that, right? Then again, you apparently need to write `&**self`?